src/sentry/static/sentry/app/components/events/interfaces/generic.jsx

```diff
@@ -0,0 +1,81 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import SentryTypes from '../../../proptypes';
+
+import GroupEventDataSection from '../eventDataSection';
+import KeyValueList from './keyValueList';
+import {t} from '../../../locale';
+import {objectToArray} from '../../../utils';
+
+function getView(view, data) {
+  switch (view) {
+    case 'report':
+      return <KeyValueList data={objectToArray(data)} isContextData={true} />;
+    case 'raw':
+      return <pre>{JSON.stringify({'csp-report': data}, null, 2)}</pre>;
+    default:
+      throw new TypeError(`Invalid view: ${view}`);
```

Comment:
Does this component's parent have an ErrorBoundary?

Reply:
Dont think so but htis is the same way CSP works.

I was thinking about how we should just create an 'Interface' base component that all of these extend. It'd make life a lot easier and we could enforce some behavior a bit better.